### PR TITLE
executor: add partition information for tikv_region_status

### LINKED ISSUE #38437: Add partition_id column on the information_schema.tikv_region_status.
## Feature Request

**Is your feature request related to a problem? Please describe:**

We can't see the related between TiKV region and partition on the information_schema.tikv_region_stauts.
**Describe the feature you'd like:**

Add paration_id or partition_name on the information_schema.tikv_region_status.

**Describe alternatives you've considered:**

Currently, we need TiDB API to see it

**Teachability, Documentation, Adoption, Migration Strategy:**

```
MySQL [test]> select tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v6.2.0-devtier
Edition: Community
Git Commit Hash: 5d17be92cc914c64c0bb4f17e875ba373364f1f2
Git Branch: release-6.2-dev-tier
UTC Build Time: 2022-08-24 01:19:35
GoVersion: go1.19
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: tikv
1 row in set (0.22 sec)

MySQL [test]> show create table t3\G
*************************** 1. row ***************************
       Table: t3
Create Table: CREATE TABLE `t3` (
  `id` int(11) NOT NULL,
  `name` varchar(10) DEFAULT NULL,
  PRIMARY KEY (`id`) /*T![clustered_index] CLUSTERED */,
  KEY `name_idx` (`name`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
PARTITION BY HASH (`id`) PARTITIONS 5
1 row in set (0.21 sec)

MySQL [test]> desc information_schema.tikv_region_status;
+---------------------------+-------------+------+------+---------+-------+
| Field                     | Type        | Null | Key  | Default | Extra |
+---------------------------+-------------+------+------+---------+-------+
| REGION_ID                 | bigint(21)  | YES  |      | NULL    |       |
| START_KEY                 | text        | YES  |      | NULL    |       |
| END_KEY                   | text        | YES  |      | NULL    |       |
| TABLE_ID                  | bigint(21)  | YES  |      | NULL    |       |
| DB_NAME                   | varchar(64) | YES  |      | NULL    |       |
| TABLE_NAME                | varchar(64) | YES  |      | NULL    |       |
| IS_INDEX                  | tinyint(1)  | NO   |      | 0       |       |
| INDEX_ID                  | bigint(21)  | YES  |      | NULL    |       |
| INDEX_NAME                | varchar(64) | YES  |      | NULL    |       |
| EPOCH_CONF_VER            | bigint(21)  | YES  |      | NULL    |       |
| EPOCH_VERSION             | bigint(21)  | YES  |      | NULL    |       |
| WRITTEN_BYTES             | bigint(21)  | YES  |      | NULL    |       |
| READ_BYTES                | bigint(21)  | YES  |      | NULL    |       |
| APPROXIMATE_SIZE          | bigint(21)  | YES  |      | NULL    |       |
| APPROXIMATE_KEYS          | bigint(21)  | YES  |      | NULL    |       |
| REPLICATIONSTATUS_STATE   | varchar(64) | YES  |      | NULL    |       |
| REPLICATIONSTATUS_STATEID | bigint(21)  | YES  |      | NULL    |       |
+---------------------------+-------------+------+------+---------+-------+
17 rows in set (0.22 sec)

MySQL [test]> select * from information_schema.tikv_region_status where TABLE_NAME = 't3';
+-----------+--------------------------------------------------------+---------+----------+---------+------------+----------+----------+------------+----------------+---------------+---------------+------------+------------------+------------------+-------------------------+---------------------------+
| REGION_ID | START_KEY                                              | END_KEY | TABLE_ID | DB_NAME | TABLE_NAME | IS_INDEX | INDEX_ID | INDEX_NAME | EPOCH_CONF_VER | EPOCH_VERSION | WRITTEN_BYTES | READ_BYTES | APPROXIMATE_SIZE | APPROXIMATE_KEYS | REPLICATIONSTATUS_STATE | REPLICATIONSTATUS_STATEID |
+-----------+--------------------------------------------------------+---------+----------+---------+------------+----------+----------+------------+----------------+---------------+---------------+------------+------------------+------------------+-------------------------+---------------------------+
|      8001 | 7480000000000000FF525F698000000000FF000008FB00000000FB |         |       90 | test    | t3         |        1 |        1 | name_idx   |              1 |            71 |             0 |    1517680 |              173 |           837084 | NULL                    |                      NULL |
|      8001 | 7480000000000000FF525F698000000000FF000008FB00000000FB |         |       90 | test    | t3         |        0 |     NULL | NULL       |              1 |            71 |             0 |    1517680 |              173 |           837084 | NULL                    |                      NULL |
|      8001 | 7480000000000000FF525F698000000000FF000008FB00000000FB |         |       90 | test    | t3         |        0 |     NULL | NULL       |              1 |            71 |             0 |    1517680 |              173 |           837084 | NULL                    |                      NULL |
|      8001 | 7480000000000000FF525F698000000000FF000008FB00000000FB |         |       90 | test    | t3         |        0 |     NULL | NULL       |              1 |            71 |             0 |    1517680 |              173 |           837084 | NULL                    |                      NULL |
|      8001 | 7480000000000000FF525F698000000000FF000008FB00000000FB |         |       90 | test    | t3         |        0 |     NULL | NULL       |              1 |            71 |             0 |    1517680 |              173 |           837084 | NULL                    |                      NULL |
|      8001 | 7480000000000000FF525F698000000000FF000008FB00000000FB |         |       90 | test    | t3         |        0 |     NULL | NULL       |              1 |            71 |             0 |    1517680 |              173 |           837084 | NULL                    |                      NULL |
+-----------+--------------------------------------------------------+---------+----------+---------+------------+----------+----------+------------+----------------+---------------+---------------+------------+------------------+------------------+-------------------------+---------------------------+
6 rows in set (0.26 sec)
```

### PR DESCRIPTION
**Pull request by jiyfhust**
### What problem does this PR solve?

Issue Number: close #38437 #29792

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by jiyfhust** [comment]
hi @mjonss, could you help review this pr?

File: executor/infoschema_reader.go [hunk 1]
```diff
@@ -1831,18 +1831,27 @@ func (e *memtableRetriever) setNewTiKVRegionStatusCol(region *helper.RegionInfo,
 		} else {
 			row[6].SetInt64(0)
 		}
+		if table.IsPartition {
+			row[9].SetInt64(1)
+			row[10].SetInt64(table.Partition.ID)
```
Comment (zimulala) [inline executor/infoschema_reader.go:1836]: How about the partition table that has multiple partitions? Such as 2 partitions. Maybe we can refer to the partitions table: https://docs.pingcap.com/tidb/stable/information-schema-partitions. And the code is here: https://github.com/pingcap/tidb/blob/master/executor/infoschema_reader.go#L1087

**Review by zimulala — COMMENTED** [review]
And please add UTs.

File: executor/infoschema_reader.go (hunk 1 above)
Comment (jiyfhust) [reply executor/infoschema_reader.go:1836]: @zimulala "How about the partition table that has multiple partitions? Such as 2 partitions. ".
Is it mean "SUBPARTITION_NAME"? 
There is already  "PARTITON_ID" and "PARTITION_NAME " to distinguish between partitions.

**Review by mjonss — APPROVED** [review]
Apart from a spelling error, it looks good to me.

File: executor/infoschema_reader.go [hunk 2]
```diff
@@ -1836,18 +1836,27 @@ func (e *memtableRetriever) setNewTiKVRegionStatusCol(region *helper.RegionInfo,
...
+			row[11].SetString(table.Partition.Name.O, mysql.DefaultCollationName)
+		} else {
+			row[9].SetInt64(0)
+		}
+
```
Comment (mjonss) [inline executor/infoschema_reader.go:1846]: I think this is the reason for the error:
executor/infoschema_reader.go:1828:18: empty-lines: extra empty line at the end of a block (all_revive)
```suggestion
```

**Review by Defined2014 — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Comment by Defined2014** [comment]
@jiyfhust Please merge master, and update this PR, something wrong with this PR/ci now.

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**